p2p/peer_tracker.go

```diff
@@ -179,6 +206,17 @@ func (p *peerTracker) gc() {
 					delete(p.trackedPeers, id)
 				}
 			}
+
```

Comment:
i'd take a copy of p.trackedPeers after here, then unlock the mutex, then range over the copy to get addr info and store the peers

Reply:
done